### ISSUE TITLE
Make each build directory a releasable package with a single `@click` cli entrypoint

### ISSUE THREAD

**Issue by cisaacstern** [wildlife-dynamics/ecoscope-workflows#248]
#225 bundles all workflow artifacts into a build directory with a self-describing (pixi) environment. This is a great step forward but has two non-insignificant challenges regarding portability for invocation from a to-be-implemented microservice (xref #246):

1. **Installation**. If these build directories are not released packages, but rather just directories with a well-known structure, then we need to know how to "fetch" them from a storage backend (local, GitHub, etc.), and then also how to install their dependencies (using pixi). These are both doable but are add more complexity than if we could simply just "pixi install" the workflow and all its dependencies (in which case the package manager handles fetching and dependencies installation for us in a single command, and we get to offload this responsibility from the microservice).
2. **Invocation**. If these build directories are not released packages, then in order to invoke a particular style of script we need to know the _name of the actual file_ in the directory, which then becomes a potentially a brittle interface, as multiple parts of the system need to know and correctly handle that naming convention. Instead, if we abstract invocation behind a single CLI, that becomes a much cleaner interface for the 

Note that we can maintain the direct-invocation behavior for ease of debugging and separability of the scripts from the proposed CLI by simply moving the script logic into a `main` function and calling that either directly (as shown below) or from the unified CLI.

```python
def main():
    # Script logic here
    print("Script 1 executing")

if __name__ == '__main__':
    main()
```

My proposed implementation is to change the script jinja templates to use a `main()` function (very simple to do), as shown above, and then add the following to the generated workflow artifacts:

- [x] A `@click` cli entrypoint that can call these `main()` functions from a unified interface
- [x] A `pyproject.toml` that sets this cli as an entrypoint
- [ ] A `rattler-build-recipe.yaml` that allows build
- [ ] In existing `pixi.toml`, add a pixi feature with tasks for building and pushing the workflow as a conda package to prefix.dev

Then in the script runner microservice, we can build a container that installs the released workflow conda package from a particular version, and the FastAPI service running in that container can invoke the workflow directly as an executable (rather than as a script).

**Comment by cisaacstern** [comment]
> - A rattler-build-recipe.yaml that allows build
> - In existing pixi.toml, add a pixi feature with tasks for building and pushing the workflow as a conda package to prefix.dev

Going to address this as part of #264

**Comment by cisaacstern** [comment]
closed by #262
